src/libsyntax/attr.rs

```diff
@@ -77,6 +81,93 @@ pub fn is_used(attr: &Attribute) -> bool {
     })
 }
 
+pub trait AttrMetaListItemMethods {
```

Comment:
It seems like adding `literal` to `AttrMetaMethods` will be a lot cleaner than defining an entire duplication of `AttrMetaMethods` to add `literal`. It will also make things more uniform in the other parts of the compiler.